rust/integration-tests/src/sdk/mock_grpc.rs

```diff
@@ -0,0 +1,89 @@
+use aptos_protos::indexer::v1::{
+    raw_data_server::{RawData, RawDataServer},
+    GetTransactionsRequest, TransactionsResponse,
+};
+use futures::Stream;
+use std::pin::Pin;
+use tonic::{Request, Response, Status};
+
+#[derive(Default)]
+pub struct MockGrpcServer {
+    pub transactions: Vec<TransactionsResponse>,
+    pub chain_id: u64,
+}
+
+type ResponseStream = Pin<Box<dyn Stream<Item = Result<TransactionsResponse, Status>> + Send>>;
+
+#[tonic::async_trait]
+impl RawData for MockGrpcServer {
```

Comment:
Instead of mocking the GRPC stream directly, might be easier to mock TransactionStream step's `poll` to return each transaction in the test. Then mock `should_continue_polling = false` to stop the processor.

Reply:
You can mock a sequence of values which is pretty cool. https://github.com/aptos-labs/aptos-indexer-processors-sdk/blob/343a2d01dce9f8bca76b90d506f61c1a07943e4b/aptos-indexer-processors-sdk/sdk/src/common_steps/transaction_stream_step.rs#L242

Reply:
ohh that's intersting, let me look into it